File: IT/Oggetti/Esercizi/Java/Esercizio24.java
/* file Esercizio24.java */

class Point2d {
	private double x;
	private double y;

	public Point2d(double x, double y) {
		this.setX(x);
		this.setY(y);
	}

	public double getX() {
		return this.x;
	}

	public void setX(double x) {
		this.x = x;
	}

	public double getY() {
		return this.y;
	}

	public void setY(double y) {
		this.y = y;
	}

	@Override
	public String toString() {
		return "(" + this.x + ", " + this.y + ")";
	}
}

class Segment {
	private Point2d a;
	private Point2d b;

	public Segment(Point2d a, Point2d b) {
		this.setA(a);
		this.setB(b);
	}

	public Point2d getA() {
		return this.a;
	}

	public void setA(Point2d a) {
		this.a = a;
	}

	public Point2d getB() {
		return this.b;
	}

	public void setB(Point2d b) {
		this.b = b;
	}

	public double length() {
		return Math.sqrt(Math.pow((this.a.getX() - this.b.getX()), 2.0) + Math.pow((this.a.getY() - this.b.getY()), 2.0));
	}
}

public class Esercizio24 {
	public static void main(String [] args) {
		Point2d o = new Point2d(0, 0);
		Point2d p = new Point2d(1, 2);
		Segment s = new Segment(o, p);

		System.out.println(o + " " + p);
		System.out.println("Distanza tra i due punti: " + String.format("%.2f", s.length()) + " centimetri.");
	}
}

File: IT/Oggetti/Esercizi/Python/esercizio24.py
# file esercizio24.py

from __future__ import print_function
from math import pow, sqrt

class Point2d(object):
	def __init__(self, x, y):
		self.__x = x
		self.__y = y

	def getX(self):
		return self.__x

	def setX(self, x):
		self.__x = x

	def getY(self):
		return self.__y

	def setY(self, y):
		self.__y = y

	def __str__(self):
		return "(x={}, y={})".format(self.__x, self.__y)

	x = property(getX, setX)
	y = property(getY, setY)

class Segment(object):
	def __init__(self, a, b):
		self.__a = a
		self.__b = b

	def getA(self):
		return self.__a

	def setA(self, a):
		self.__a = a

	def getB(self):
		return self.__b

	def setB(self, b):
		self.__b = b

	def length(self):
		return sqrt((self.a.x - self.b.x)**2 + (self.a.y - self.b.y)**2)

	a = property(getA, setA)
	b = property(getB, setB)

o = Point2d(0, 0)
p = Point2d(1, 2)
s = Segment(a=o, b=p)

print("O = {}, P = {}".format(o, p))
print("Distanza tra i due punti: {:.2f} centimetri.".format(s.length()))

